### PR TITLE
Animated dust-on-death

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
I am.. inevitable.

This PR adds animations to dust on death. When you activate the implant, you now get dusted out of reality.

6rQLqJVisI

*Rough code and displacement map from Absolucy* 

Nerd shit explanation:
Each pixel on the displacement map is a unique colour. This is then applied as a filter on the mob image and the pixels are tracked. When pixels move on the displacement map, they update on the mob sprite as having moved.

## Why It's Good For The Game
It looks fucking sick/

## Changelog
:cl: AffectedArc07
imageadd: Dust implants are now gloriously animated
/:cl:

Yes I know we're in a feature freeze. I have *plenty* of refactors to cite.

### CODE REVIEW

**Comment by hal9000PR** [comment]
Is this only for dust implants or all dusting effects?

**Comment by AffectedArc07** [comment]
> Is this only for dust implants or all dusting effects?

Anything which calls `dust()` on the mob will have this happen. I used the implants purely as an example.

**Comment by buffyuwu** [comment]
very cool

**Comment by Sirryan2002** [comment]
does this mean each mob doesn't need its own dust animation? cause if so, fucking awesome.

**Comment by AffectedArc07** [comment]
> does this mean each mob doesn't need its own dust animation? cause if so, fucking awesome.

This is currently only scoped at the human level, but we can scope it lower if needed.

**Comment by Adrer** [comment]
Mr Affected
I don't feel so good

**Comment by hal9000PR** [comment]
> Mr Affected

Mr Arc was staring you in the face and you wasted it.

File: code/game/objects/effects/misc.dm [hunk 1]
```diff
@@ -129,3 +129,15 @@
...
 	if(!ismob(loc))
 		return INITIALIZE_HINT_QDEL
+
+/obj/effect/snap_rt
+	icon = 'icons/effects/filters.dmi'
+	icon_state = "nothing"
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	plane = FLOOR_PLANE
+
+/obj/effect/snap_rt/New(_loc, id)
```
Comment (farie82) [inline code/game/objects/effects/misc.dm:139]: is there a reason this is using `New` over `Initialize`?

File: code/game/objects/effects/misc.dm [hunk 2]
```diff
@@ -129,3 +129,15 @@
...
+	..()
+	loc = _loc
```
Comment (farie82) [inline code/game/objects/effects/misc.dm:141]: This is done by byond itself right? The first param given when newing an object will be the loc of the `atom`

File: code/modules/mob/living/carbon/human/death.dm [hunk 3]
```diff
@@ -49,25 +49,20 @@
...
-	QDEL_IN(src, 15)
+	QDEL_IN(src, 20)
 	return TRUE
 
 /mob/living/carbon/human/dust_animation()
-	var/atom/movable/overlay/animation = null
+	// Animate them being dusted out of existence
+	var/obj/effect/snap_rt/dust_effect = new(loc, UID())
+	filters += filter(type="displace", size=256, render_source="*snap[UID()]")
+	animate(src, alpha=0, time=20, easing=(EASE_IN | SINE_EASING))
```
Comment (farie82) [inline code/modules/mob/living/carbon/human/death.dm:62]:
```suggestion
	filters += filter(type = "displace", size = 256, render_source = "*snap[UID()]")
	animate(src, alpha = 0, time = 2 SECONDS, easing = (EASE_IN | SINE_EASING))
```

Comment (farie82) [inline code/modules/mob/living/carbon/human/death.dm:62]: Question regarding this code btw. This is what actually does the effect right? The `/obj/effect/snap_rt` object itself is just a holder with no behaviour right?
If so wouldn't it make more sense to move this logic to the effect itself?

**Review by farie82 — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/misc.dm (hunk 1 above)
Comment (AffectedArc07) [reply code/game/objects/effects/misc.dm:139]: I was going to make this a datum then realised that doesnt work

File: code/game/objects/effects/misc.dm (hunk 2 above)
Comment (AffectedArc07) [reply code/game/objects/effects/misc.dm:141]: I was going to make this a datum then realised that doesnt work

File: code/modules/mob/living/carbon/human/death.dm (hunk 3 above)
Comment (AffectedArc07) [reply code/modules/mob/living/carbon/human/death.dm:62]: Probably. Ill try that tomorrow.

Comment (AffectedArc07) [reply code/modules/mob/living/carbon/human/death.dm:62]:
```dm
	var/mutable_appearance/MA = mutable_appearance('icons/effects/filters.dmi', "snap3", ABOVE_ALL_MOB_LAYER, FLOOR_PLANE)
	MA.render_target = "dusting_[UID()]"
	filters += filter(type="displace", size=256, render_source="dusting_[UID()]")
```

Does not work

Comment (farie82) [reply code/modules/mob/living/carbon/human/death.dm:62]: I honestly don't exactly know how `render_target` works for byond.
But I meant to say to move the logic itself to `/obj/effect/snap_rt` or a global proc which does the animation for you. Currently, you can spawn `/obj/effect/snap_rt` with the proper params and it does nothing on its own.

Comment (Vi3trice) [reply code/modules/mob/living/carbon/human/death.dm:62]: Have you tried with the asterisk? Render targets that start with it are handled differently.

Comment (AffectedArc07) [reply code/modules/mob/living/carbon/human/death.dm:62]: * dont help it

**Review by dearmochi — APPROVED** [review]

**Pull request merged**